lib/cuckoo/common/abstracts.py

```diff
@@ -107,42 +106,44 @@ class Machinery:
     # Default label used in machinery configuration file to supply virtual
     # machine name/label/vmx path. Override it if you dubbed it in another
     # way.
-    LABEL = "label"
+    LABEL: str = "label"
+
+    # This must be defined in sub-classes.
+    module_name: str
 
     def __init__(self):
-        self.module_name = ""
         self.options = None
         # Database pointer.
-        self.db = Database()
-        # Machine table is cleaned to be filled from configuration file
-        # at each start.
-        self.db.clean_machines()
+        self.db: _Database = Database()
+        self.set_options(self.read_config())
+
+    def read_config(self) -> None:
+        return Config(self.module_name)
 
-    def set_options(self, options: dict):
+    def set_options(self, options: dict) -> None:
         """Set machine manager options.
         @param options: machine manager options dict.
         """
         self.options = options
+        mmanager_opts = self.options.get(self.module_name)
+        if not isinstance(mmanager_opts["machines"], list):
+            mmanager_opts["machines"] = str(mmanager_opts["machines"]).strip().split(",")
```

Comment:
This is not with dynamic machinery like AZ and AWS, hence creating a problem with parsing this.

Reply:
Would you mind trying out this patch?
[machine_lock.patch.txt](https://github.com/kevoreilly/CAPEv2/files/14909811/machine_lock.patch.txt)

Reply:
Seem to initialize fine, thanks for patching !

Reply:
DEBUG: # Active analysis: 1; # Pending Tasks: 0; # Specific Pending Tasks: {}; # Available Machines: 10; # Available Specific Machines: {'win10x64': 9, 'windows': 9, 'ub20x64': 1, 'linux': 1}; # Locked Machines: 1; # Specific Locked Machines: {'win10x64': 1, 'windows': 1}; # Total Machines: 11; # Lock value: 2/2 It seem the lock value doesn't follow correctly the number of available machines. I guess this is a work of progress because of this : ```
 #Here be dragons! Making these calls on the ScalingBoundedSemaphore is not
                # thread safe.
```

Reply:
Could you try adding some debug logging in `update_limit()` and `check_for_starvation()` to make sure that they are getting called every 10 seconds (or whatever you have `scaling_semaphore_update_timer` set to) and make sure they're getting called with accurate parameter values?

Reply:
So the thread in question which invoke `update_limit() `and `check_for_starvation()` is just never started (missing a start()), so this is the source of this problem !

Reply:
I've created https://github.com/kevoreilly/CAPEv2/pull/2055 to start the thread.